ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl.hs

```diff
@@ -127,10 +127,14 @@ openDBInternal args launchBgTasks = do
           LgrDB.decorateReplayTracer
             immutableDbTipPoint
             (contramap TraceLedgerReplayEvent tracer)
+    -- TODO: at this point the LgrDB.changelogLock is not created, so the ledger
+    -- DB initialization from disk can flush freely.
     (lgrDB, replayed) <- LgrDB.openDB argsLgrDb
                             lgrReplayTracer
                             immutableDB
                             (Query.getAnyKnownBlock immutableDB volatileDB)
```

Comment:
Nick's intuition: the flush-lock could be part of the `ChainDB` as opposed to the `LgrDB`.

Reply:
The `LgrDB` gets to pretend the world is single threaded.

Reply:
> The LgrDB gets to pretend the world is single-threaded (separation of concerns).

I think this is not true: there are STM operations in `LgrDB` already.

Reply:
Sorry I don't get this :)